cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -1160,13 +1145,25 @@ updateCreateStakedOutputTemplate
     unLovelace (Lovelace coin) = fromIntegral coin
 
 writeFileGenesis
-  :: ToJSON genesis
-  => FilePath
-  -> genesis
-  -> ExceptT ShelleyGenesisCmdError IO ()
-writeFileGenesis fpath genesis =
+  :: FilePath
+  -> WriteFileGenesis
+  -> ExceptT ShelleyGenesisCmdError IO (Crypto.Hash Crypto.Blake2b_256 ByteString)
+writeFileGenesis fpath genesis = do
   handleIOExceptT (ShelleyGenesisCmdGenesisFileError . FileIOError fpath) $
-    LBS.writeFile fpath (Aeson.encode genesis)
+    BS.writeFile fpath content
+  return $ Crypto.hashWith id content
+  where
+    content = case genesis of
+       WritePretty a -> LBS.toStrict $ encodePretty a
+       WriteCanonical a -> LBS.toStrict
+          . renderCanonicalJSON
+          . either (error "error parsing json that was just encoded!?") identity
+          . parseCanonicalJSON
+          . canonicalEncodePretty $ a
+
+data WriteFileGenesis where
```

Comment:
It would be better to avoid the creation of another data type. We can simply have a function defined in `Cardano.CLI.Byron.Genesis` that does the correct JSON encoding for the Byron genesis and define a function here specifically for the Shelley genesis. Then just call those where appropriate.

Reply:
I do not know why Byron was special at first. I just reordered the existing code.
There is a single function that handles writing Byron and Shelley genesis files.
The data type is local and not exported. It makes the arguments of the function more explicit.
The definition of the data type itself is only three lines of code.
Why not create that data type ?

Reply:
Actually this is fine as its not exported :+1: